dqc_us_rules/dqc_us_0033_0036.py

```diff
@@ -117,17 +121,24 @@ def _doc_period_end_date_check(model_xbrl):
             # If the DocumentPeriodEndDate context check doesn't fire,
             # we will check all dei fact context end dates against it.
             for fact in fact_group:
-                if ((fact.contextID in not_valid_dped or
-                     fact.context is None or
+                if ((fact.context is None or
                      fact.context.endDatetime is None or
                      fact.concept.periodType != 'duration'
                      )):
                     continue
 
-                if context_eop_date != dateunionDate(
-                    fact.context.endDatetime,
-                    subtractOneDay=True
-                ):
+                should_break = False
```

Comment:
Only a suggestion but you could refactor the inner loop into a function and return True when you break. Then you could also document the function and make the code more readable.

ie.
def function(fact_items):
     for fact_axis, fact_dim_value in fact.context.segDimValues.items():
           if fact_dim_value.memberQname.localName in not_valid_dped:
               return True
     return False

if function():
    continue